### PR TITLE
Removal of external site logic

### LINKED ISSUE #638: Remove external site access code/permission logic from core
External site access is in a semi-maintained state. Some permissions logic requires the user's `site_id` to be checked, some doesn't. We currently do not support access from external sites. This should be removed until there is a more concentrated effort to implement this feature.

Todo:
  - remove any reference to `site` on all permissions lists on all containers
  - remove references to `site` from permissions checking logic ( most of which is in `api/auth/`)
  - remove logic from `BaseHandler()` and `centralclient.py` (or maybe all of `centralclient.py`) that references the cross-site logic

### PR DESCRIPTION
**Pull request by hkethi002**
Fixes #638 
### Breaking Changes
- SiteId is no longer an accepted input for permissions
### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import webapp2
 import webapp2_extras.routes
 
-from .centralclient               import CentralClient
+# from .centralclient               import CentralClient
```
Comment (nagem) [inline api/api.py]: You can remove this line 👍

File: bin/database.py [hunk 2]
```diff
@@ -1054,6 +1054,65 @@ def upgrade_to_29():
...
+    process_cursor(cursor, upgrade_to_30_acquisitions_closure)
+
+    cursor = config.db.groups.find({'roles.site': {'$exists': True}})
+    process_cursor(cursor, upgrade_to_30_groups_closure)
+
+    cursor = config.db.projects.find({'permissions.site' : {'$exists' : True}})
+    process_cursor(cursor, upgrade_to_30_projects_closure)
+
+    cursor = config.db.sessions.find({'permissions.site': {'$exists' : True}})
+    process_cursor(cursor, upgrade_to_30_sessions_closure)
```
Comment (nagem) [inline bin/database.py]: We should definitely rebase this branch off master when the timestamp fix lands and maybe bring in the roles->permissions rename changes as well.

File: raml/schemas/definitions/permission.json [hunk 3]
```diff
@@ -16,7 +16,7 @@
     },
     "permission-output-default-required":{
```
Comment (nagem) [inline raml/schemas/definitions/permission.json]: We'll want to remove site from the `permission` schema definition.

**Comment by nagem** [comment]
Only issue I saw was noted in the comment above (modifying the json schema to prevent clients from sending the site when modifying permissions lists). Unfortunately that makes this a breaking change to the API.

File: bin/database.py [hunk 4]
```diff
@@ -1115,6 +1115,20 @@ def upgrade_to_31():
...
+            del permission_['site']
+    result = config.db[coll].update_one({'_id': coll_item['_id']}, {'$set': {'permissions' : permissions}})
+    if result.modified_count == 0:
+        return "Failed to remove site field"
+    return True
+
+def upgrade_to_32():
+    for coll in ['acquisitions', 'groups', 'projects', 'sessions']:
+        cursor = config.db[coll].find({'permissions.site': {'$exists': True}})
+        process_cursor(cursor, upgrade_to_32_closure, context = coll)
```
Comment (nagem) [inline bin/database.py]: Here we can also remove the "sites" collection, as well as remove it's creation in `config.py`.

Comment (nagem) [inline bin/database.py]: @ryansanford you never use the `sites` collection in Mongo, correct?

Comment (nagem) [inline bin/database.py]: Currently it should only hold the current site's `id`, `name`, and `site_url` which are all available in the config.

**Comment by kofalt** [comment]
> +135 −368

:tada: :tada: :tada:

**Pull request merged**